### DISCUSSION TITLE
Playwright-allure. View trace directly from the report as playwright reporter

### DISCUSSION THREAD

**Discussion by Amirdjan94** [allure-framework/allure2#1934] [Questions & Support] [answered]
The allure report does not have enough integrated view trace, as in playwright reporter.
Are there any plans to implement this functionality?

**Comment by EldoradoEU** [comment]
any update?

**Comment by danielalmeidagoncalves** [comment]
This would be nice indeed.

**Comment by marliz97** [comment]
+1

**Comment by baev** [comment]
This issue is on the radar for both Allure Report & Allure TestOps. I'll post the update here as soon as we'll have an ETA

**Reply by o-williams** [reply]
Also interested in this

**Reply by caseyjhol** [reply]
Anything the community can do to help move this along? I'd be interested in working on a PR, but don't want to do double work if someone else has already started on this.

**Reply by Maximus1285** [reply]
Any update here?

**Reply by ofirlupa** [reply]
Has anyone been able to put in their allure report a direct link to open the playwright's trace.zip??

**Comment by DrMarkDunne** [comment]
@baev any update here? thanks.

**Comment by ofirlupa** [comment]
any news ?

**Comment by baiyyee** [comment]
+1

**Comment by baev** [comment]
The feature is merged https://github.com/allure-framework/allure2/pull/2964. Will be available in the next release. 

BTW [Allure 3](https://github.com/allure-framework/allure3) already supports it!

**Reply by baev** [reply]
PS the release is delayed due to OSSRH EOL

> The OSSRH service will reach end-of-life on [June 30th, 2025](https://central.sonatype.org/news/20250326_ossrh_sunset/)

**Reply by baev** [reply]
A [link](https://allure-framework.github.io/allure3-demo/awesomeE2E/#c43ff0399f3c0032aa64b1b527bcbb29) for you to check out Allure 3 with the Playwright Trace

**Reply by ofirlupa** [reply]
@baev 

Nice !! very nice 

another thing, allow to open this from the zip code. 

i work on python playwright and grab the trace when i have fail :

`@pytest.fixture(scope="function")
def page(browser_context, request):
    page = browser_context.new_page()
    browser_context.tracing.start(screenshots=True, snapshots=True, sources=True)
    failed_before = request.session.testsfailed
    yield page
    if request.session.testsfailed != failed_before:
        test_name = request.node.name
        trace_path = "trace.zip"
        browser_context.tracing.stop(path=trace_path)
        allure.attach.file(trace_path,name=test_name + ".zip",attachment_type="application/zip")
        screenshot = browser_context.pages[0].screenshot()
        allure.attach(body=screenshot, name="FailShot", attachment_type=allure.attachment_type.PNG)
    else:
        browser_context.tracing.stop()
    page.close()`

**Reply by baev** [reply]
Please create a separate discussion and I'll happy to help! (ATM not sure what is the question)

**Comment by arekon7** [comment]
Does Playwright Trace viewer implemented inside Allure report already or not?

**Reply by baev** [reply, marked as answer]
yes, it is supported in both Allure 3 and Allure 2 (since [2.35.0](https://github.com/allure-framework/allure2/releases/tag/2.35.0))

**Reply by arekon7** [reply]
Thank you for the answer, is there any documentation or example how to setup trace viewer inside Allure report? using: "allure-playwright": "^3.4.5"

**Reply by baev** [reply]
Documentation is WIP, as for now you can refer to https://github.com/allure-framework/allure-js/pull/1274

In short, all you need is to attach trace file with `application/vnd.allure.playwright-trace` content type
